test/unit/BPool.t.sol

```diff
@@ -83,13 +83,19 @@ abstract contract BasePoolTest is Test, BConst, Utils, BMath {
   function _expectRevertByReentrancy() internal {
     // Assert that the contract is accessible
     assertEq(bPool.call__getLock(), _MUTEX_FREE);
-
     // Simulate ongoing call to the contract
     bPool.call__setLock(_MUTEX_TAKEN);
 
     vm.expectRevert(IBPool.BPool_Reentrancy.selector);
   }
 
+  function _expectSetReentrancyLock() internal {
+    // Assert that the contract is accessible
+    assertEq(bPool.call__getLock(), _MUTEX_FREE);
+    // Expect reentrancy lock to be set
+    bPool.expectCall__setLock(_MUTEX_TAKEN);
+  }
```

Comment:
mhhhh i think i'd remove the internal and just do `bPool.expectCall__setLock(_MUTEX_TAKEN);` on each method tested

also would call them `test_Call_SetLock`

Reply:
i correct myself:
- i don't see the point in checking that the current lock is FREE
- but, i do find useful, since you made this internal, to maintain it, because in the event that we wanna check it in another way, we already have that abstraction made

Reply:
I wanted to replicate the behaviour of `_expectRevertReentrancy`, it'd for example be useful as a sanity check if the lock is already set by another not-properly-isolated call